client/src/interfaces/position.ts

```diff
@@ -0,0 +1,57 @@
+export interface IPosition {
+  x: number;
+  y: number;
+}
+
+export class TilePositon implements IPosition {
+  constructor(
+    public x: number,
+    public y: number,
+  ) {}
+  /**
+   * Creates a `TilePositon` from an object with x, and y
+   */
+  public static fromXY = ({ x, y }: {x: number; y: number}): TilePositon => new TilePositon(x, y);
```

Comment:
in general, I try to avoid static functions if I can help it; these would go to a separate helper service or something.